.github/workflows/codeql.yml

```diff
@@ -0,0 +1,60 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
+    paths-ignore:
+      - '**/*.md'
+      - '**/*.txt'
+      - '**/*.yaml'
+      - '**/*.json'
+      - '**/node_modules'
+      - '**/*.test.js'
+      - '**/*.test.ts'
+  pull_request:
+    branches: [ "master" ]
```

Comment:
Not recommended to pin branch for pr event. Please remove this line so it scans against any branches.